src/glyph_item_iter.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::translate::*;
+use pango_sys;
+use GlyphItem;
+
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct GlyphItemIter(Boxed<pango_sys::PangoGlyphItemIter>);
+
+    match fn {
+        copy => |ptr| pango_sys::pango_glyph_item_iter_copy(mut_override(ptr)),
+        free => |ptr| pango_sys::pango_glyph_item_iter_free(ptr),
+        init => |_ptr| (),
+        clear => |_ptr| (),
+        get_type => || pango_sys::pango_glyph_item_iter_get_type(),
+    }
+}
+
+impl GlyphItemIter {
+    pub fn init_end(glyph_item: &mut GlyphItem) -> Option<GlyphItemIter> {
+        unsafe {
+            let mut iter = GlyphItemIter::uninitialized();
+            let ret = from_glib(pango_sys::pango_glyph_item_iter_init_end(
+                iter.to_glib_none_mut().0,
+                glyph_item.to_glib_none_mut().0,
+                //Text seems ignored and item's used
+                "".to_glib_none().0,

```

Comment:
This is wrong. This must be the text corresponding to the glyph item, or otherwise you get invalid memory accesses. No idea how to enforce that :)

See https://gitlab.gnome.org/GNOME/pango/blob/050ebae3a0340e37187319273b5ab443cbe1f0fd/pango/pango-glyph-item.c#L287-288 for where this is used for example.

Reply:
Strange but for me this works fine for Layout

Reply:
I don't know, I can only tell you what the code looks like :) It's taking the text pointer without any checks and then adds offsets calculated from the glyph item on it without checking that it's actually in the valid range of the text.